### PR TITLE
*: fix sysvar inconsistencies/potential bugs

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Problem Summary:

This fixes two inconsistencies/potential bugs with sysvars:

1. There are a number of Int/Unsigned sysvars that have a range which includes a `MaxValue` greater than the internal value which stores the setting, for example:

```
	{Scope: ScopeGlobal | ScopeSession, Name: TiDBExecutorConcurrency, Value: strconv.Itoa(DefExecutorConcurrency), Type: TypeUnsigned, MinValue: 1, MaxValue: math.MaxUint64, SetSession: func(s *SessionVars, val string) error {	{Scope: ScopeGlobal | ScopeSession, Name: TiDBExecutorConcurrency, Value: strconv.Itoa(DefExecutorConcurrency), Type: TypeUnsigned, MinValue: 1, MaxValue: math.MaxInt32, SetSession: func(s *SessionVars, val string) error {
		s.ExecutorConcurrency = tidbOptPositiveInt32(val, DefExecutorConcurrency)
		return nil
	}},
```

`Math.Uint64` is much greater than `tidbOptPositiveInt32`. The precision is lost when setting the value, not in the validation function.. which makes this behavior undefined.

2. There are a number of variables which are inconsistently named using "TIDB" (all upper), when the convention is "TiDB". Since this is just the constant name, and doesn't change the storage name, it's safe to rename them.

### What is changed and how it works?

Minor code fixes.

### Related changes

-None

### Check List 

Tests 

- Unit test (covered by existing tests)

Side effects

-None

### Release note 

- No release note

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -893,7 +893,7 @@ var defaultSysVars = []*SysVar{
 		s.CorrelationThreshold = tidbOptFloat64(val, DefOptCorrelationThreshold)
 		return nil
 	}},
-	{Scope: ScopeGlobal | ScopeSession, Name: TiDBOptCorrelationExpFactor, Value: strconv.Itoa(DefOptCorrelationExpFactor), Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxUint64, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBOptCorrelationExpFactor, Value: strconv.Itoa(DefOptCorrelationExpFactor), Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxInt32, SetSession: func(s *SessionVars, val string) error {
```
Comment (wjhuang2016) [inline sessionctx/variable/sysvar.go:896]: Can we use I64 since tidbOptInt64()?

Comment (morgo) [reply sessionctx/variable/sysvar.go:896]: It is `tidbOptInt64()` wrapped in a cast to `int`, and `s.CorrelationExpFactor` is only an int.

File: sessionctx/variable/sysvar.go [hunk 2]
```diff
@@ -983,7 +983,7 @@ var defaultSysVars = []*SysVar{
 		s.BatchCommit = TiDBOptOn(val)
 		return nil
 	}},
-	{Scope: ScopeGlobal | ScopeSession, Name: TiDBDMLBatchSize, Value: strconv.Itoa(DefDMLBatchSize), Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxUint64, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBDMLBatchSize, Value: strconv.Itoa(DefDMLBatchSize), Type: TypeUnsigned, MinValue: 0, MaxValue: math.MaxInt64, SetSession: func(s *SessionVars, val string) error {
```
Comment (wjhuang2016) [inline sessionctx/variable/sysvar.go:986]: Then why do you use MaxInt64 here?

Comment (morgo) [reply sessionctx/variable/sysvar.go:986]: It looks like an incorrect fix. I'll change it to be `math.MaxInt32` as well.

Comment (morgo) [reply sessionctx/variable/sysvar.go]: https://github.com/pingcap/tidb/pull/24964/commits/b86105a5cb1d5ec82d9e1ff0ed2a4530e4874f15

**Review by wjhuang2016 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**